src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+        if (hasAttribute(GATKSVVCFConstants.INSERTED_SEQUENCE)) {
+            final String asString = getAttributeAsString(GATKSVVCFConstants.INSERTED_SEQUENCE, null);
+            if (asString == null) {
+                return null;
+            } else {
+                return asString.getBytes();
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Returns the absolute structural variant length as recorded in the SVLEN annotation.
+     * <p>
+     *     This method always return a 0 or positive value, so for example for a deletion of 10bp
+     *     it will return 10 despite that the SVLEN annotation should be -10.
+     * </p>
+     * @return -1 if there is no SVLEN annotation, 0 or greater otherwise.
+     */
+    public int getStructuralVariantLength() {
+        if (length < 0) {
+            if (hasAttribute(GATKSVVCFConstants.SVLEN)) {
+                length = Math.abs(getAttributeAsInt(GATKSVVCFConstants.SVLEN, 0));
+            }
+        }
+        return length;
+    }
+
+    /**
+     * Returns break point intervals for this structural variant.
+     * <p>
+     *     Typically the break point would be located between to bases in the reference genome.
+     *     In that case this method will return {@code padding} bases up- and down-stream from that
+     *     inter-base position.
+     * </p>
+     * <p>
+     *     In case the padding is set two zero, since the 0-length interval is not valid, this method would
+     *     return an interval included the base just before the break-point.
+     * </p>
+     *
+     * @param padding the padding around the exact location of the break point to be included in the padding interval.
+     * @param dictionary reference meta-data.
+     * @return never {@code null}, potentially 0-length but typically at least one element.
+     */
+    public List<SimpleInterval> getBreakPointIntervals(final int padding, final SAMSequenceDictionary dictionary) {
+        Utils.nonNull(dictionary, "the input dictionary cannot be null");
+        final String contigName = getContig();
+        final int contigLength = dictionary.getSequence(contigName).getSequenceLength();
+        final StructuralVariantType type = getStructuralVariantType();
+        final int start = getStart();
+        if (type == StructuralVariantType.INS) {
+            return Collections.singletonList(new SimpleInterval(contigName,
+                    Math.max(1, padding > 0 ? (start - padding + 1) : start),
+                    Math.min(contigLength, start + padding)));
+        } else if (type == StructuralVariantType.DEL) { // must be <DEL>
+            final int length = getStructuralVariantLength();
+            return Arrays.asList(
+                    new SimpleInterval(contigName,
+                            Math.max(1, padding > 0 ? (start - padding + 1) : start) ,
+                            Math.min(contigLength, start + padding)), // left-end.
```

Comment:
It is, also I think I found a bug here... I reckon the right break end should be:  padding + end-end + padding